### ISSUE TITLE
homebrew/php/php56: Cannot find /usr/sbin/apxs

### ISSUE THREAD

**Issue by hannwong** [Linuxbrew/legacy-linuxbrew#506]
First posted by @stanley-cheung at Homebrew/homebrew-php#1926.

Moving that issue to here where it belongs.

Note that I did manage to install homebrew/php/php56 after doing:

```
sudo apt-get install apache2-dev
```

**Comment by iggyvolz** [comment]
Could you run brew gist-logs php56 and post the URL it gives you?  It's really hard to follow all of the error logs you posted, and gist-logs puts them in a consistant format and uploads them.

Taps are in a really awkward place with Linuxbrew - we can't really push changes to their tap, but cloning their tap and syncing it with Linux-specific changes makes a ton of extra work.

**Comment by sjackman** [comment]
@iggyvolz 

https://github.com/Homebrew/homebrew-php/issues/1926#issuecomment-129260824
> @ablyler Hi, Andy. If we get php56 up and running on Linuxbrew, would you be willing to merge those patches into Homebrew/homebrew-php?

> @sjackman sure, as long as they don't break homebrew on OS X. :-)

**Comment by iggyvolz** [comment]
Incidentally, I'm working on getting php installed on my linux server through linuxbrew and I can confirm the problem.

Installing --without-apache made the error go away.  It looks like php56 defaults to linking to OS X's preinstalled apache binary (in /usr/sbin/apxs), so that should be appended with an "unless OS.linux?" I believe.

I also got an error about DTrace so I had to build with phpdbg so that it wouldn't add --enable-dtrace (the two conflict and it defaults to DTrace, which I assume is a mac-only thing).  So the --enable-dtrace should be "unless OS.linux?"

Next I got configure: error: DBA: Could not find necessary header file(s), which I'm working on.

**Comment by sjackman** [comment]
@alanthing Hi, Alan. We're working on getting `homebrew/apache/httpd` building in Linuxbrew. If we get this going, would you be willing to merge those patches into `homebrew/apache`?

**Comment by jpmurray** [comment]
@sjackman any news on your fix? I'm trying here to install composer via homebrew and getting thoses errors from PHP56 install.

**Comment by sjackman** [comment]
No, sorry. I'm traveling until the new year, and I'll have a backlog of work when I return. This issue needs someone to adopt it.
